transferui/src/test/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModelTest.java

```diff
@@ -0,0 +1,298 @@
+package com.hyperwallet.android.ui.transfer.viewmodel;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TOKEN;
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TRANSFER_METHOD_CURRENCY;
+
+import android.content.Intent;
+
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.hyperwallet.android.Hyperwallet;
+import com.hyperwallet.android.model.HyperwalletError;
+import com.hyperwallet.android.model.HyperwalletErrors;
+import com.hyperwallet.android.model.TypeReference;
+import com.hyperwallet.android.model.transfer.Transfer;
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.model.user.HyperwalletUser;
+import com.hyperwallet.android.ui.transfer.repository.TransferRepository;
+import com.hyperwallet.android.ui.transfer.repository.TransferRepositoryFactory;
+import com.hyperwallet.android.ui.transfer.rule.HyperwalletExternalResourceManager;
+import com.hyperwallet.android.ui.transfer.rule.HyperwalletMockWebServer;
+import com.hyperwallet.android.ui.transfer.util.TestAuthenticationProvider;
+import com.hyperwallet.android.ui.transfer.view.CreateTransferActivity;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepository;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepositoryFactory;
+import com.hyperwallet.android.ui.user.repository.UserRepository;
+import com.hyperwallet.android.ui.user.repository.UserRepositoryFactory;
+import com.hyperwallet.android.util.JsonUtils;
+
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.stubbing.Answer;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.Date;
+
+@RunWith(RobolectricTestRunner.class)
+public class CreateTransferViewModelTest {
```

Comment:
Please add a test case to verify the default values (testCreateTransferViewModel_verifyDefaultValues maybe...) in both constructors. I've seen some comments about the default values (you can keep those), but I think a test case for the default values is important